### ISSUE TITLE
Move to Typescript 

### ISSUE THREAD

**Issue by TimMikeladze** [accounts-js/accounts#44]
Flow is nice but Typescript provides a better development  experience.

**Comment by davidyaha** [comment]
After talking a bit on slack I think this is a matter that should be discussed more on this issue.

Typescript 2.2 is great, it is AFAIK supporting the full feature set of current ecmascript and even offers some more unstable features (e.g. Decorators) as a free addition.

Flow Type on the other hand is still young. Most of it's drawback comes from tooling and lack of documentation. For example, bundling typings with the published code base is require a simple `cp` to the lib dir of the actual source files with the `*.flow.js` extension so that the flow typing tool could pick it up.

It is does however, have most issues covered. We have already invested a decent amount of time in order to have development experience with flow that is acceptable.

**Development flow scenario**
- Add a file
- Write in plain js
- Add typings
- Add `// @flow` directive at the beginning of the file
- `yarn test`
- `yarn flow:check`

Flow was initially chosen for a couple of reasons:
- None of us used it before and wanted to experiment with it.
- It allowed us to slowly implement type notations without affecting our logic
- Allows us to use the most highly adopted JS compiler/transpiler - BabelJS. 
  Which means any new language features is supported. This is a case where Typescript is some time left behind for a while (try object spread before 2.1....)

The two later points is still valid but is dwarfed by any big issue we still have with Flow.
So I suggest raise here issues we have with Flow as it comes up, so we could decide whether or not, it will be a good investment of someone's time to move into Typescript.

**TL;DR**
It will take a bunch of work to move into Typescript with all of the packages and we still have things much more important to do (like documentation..)

**Comment by ElectricCookie** [comment]
TypeScript is great as long as you can get it running, but using external typings and then also publishing a library supporting typings is a nightmare (resolving typings etc.). At least that's my experience...

**Comment by TimMikeladze** [comment]
Closed by #112
